crates/oxc_linter/src/rules/unicorn/no_process_exit.rs

```diff
@@ -0,0 +1,183 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{ast_util::is_method_call, context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-unicorn(no-process-exit): Disallow `process.exit()`.")]
+#[diagnostic(
+    severity(warning),
+    help("Only use `process.exit()` in CLI apps. Throw an error instead.")
+)]
+struct NoProcessExitDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoProcessExit;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallow `process.exit()`.
+    ///
+    /// ### Why is this bad?
+    /// Only use `process.exit()` in CLI apps. Throw an error instead.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// if (problem) process.exit(1);
+    ///
+    /// // Good
+    /// if (problem) throw new Error("message");
+    ///
+    /// #! /usr/bin/env node
+    /// if (problem) process.exit(1);
+    /// ```
+    NoProcessExit,
+    correctness
```

Comment:
is this correctness

Reply:
Or `suspicious`?

I'm less confident how to determine which categories should be applied. Are there any document for that? 👀

Reply:
hmm don't think so.

maybe this should be correctness. not really sure

Reply:
What about `restriction`?